### PR TITLE
Warm reboot: Add Table::getEntry() and Table::setEntry

### PR DESCRIPTION
**Pull request by jipanyang**
Signed-off-by: Jipan Yang <jipan.yang@alibaba-inc.com>

To support get/set individual field of a key in Table class.

### CODE REVIEW

File: common/table.cpp [hunk 1]
```diff
@@ -85,6 +85,44 @@ bool Table::get(const string &key, vector<FieldValueTuple> &values)
     return true;
 }
 
+bool Table::getEntry(const string &key, const std::string &field,  std::string &value)
+{
+    RedisCommand hget_entry;
+    hget_entry.format("HGET %s %s", getKeyName(key).c_str(), field.c_str());
+    //shget.format("HGET %s %s", key.c_str(), field.c_str());
```
Comment (lguohan) [inline common/table.cpp]: remove commented line?

File: common/table.cpp [hunk 2]
```diff
@@ -85,6 +85,44 @@ bool Table::get(const string &key, vector<FieldValueTuple> &values)
...
+
+    if (reply->type == REDIS_REPLY_NIL)
+    {
+        value.clear();
+        return false;
+    }
+
+    if (reply->type != REDIS_REPLY_STRING)
+        throw system_error(make_error_code(errc::address_not_available),
+                           "Unable to connect netlink socket");
```
Comment (qiluo-msft) [inline common/table.cpp]: >Unable to connect netlink socket [](start = 28, length = 32)

a misleading error message.

File: common/table.cpp [hunk 3]
```diff
@@ -85,6 +85,44 @@ bool Table::get(const string &key, vector<FieldValueTuple> &values)
...
+
+    value =  stripSpecialSym(reply->str);
+    return true;
+}
+
+void Table::setEntry(const string &key, const std::string &field, const std::string &value,
+                const string& /*op*/, const string& /*prefix*/)
+{
+    if (field.empty() || value.empty())
```
Comment (qiluo-msft) [inline common/table.cpp]: >field [](start = 8, length = 5)

Either field or value could be empty string

File: common/table.cpp [hunk 4]
```diff
@@ -85,6 +85,44 @@ bool Table::get(const string &key, vector<FieldValueTuple> &values)
...
+    {
+        value.clear();
+        return false;
+    }
+
+    if (reply->type != REDIS_REPLY_STRING)
+        throw system_error(make_error_code(errc::address_not_available),
+                           "Unable to connect netlink socket");
+
+    value =  stripSpecialSym(reply->str);
```
Comment (qiluo-msft) [inline common/table.cpp]: >  [](start = 12, length = 1)

One more blank?

File: common/table.h [hunk 5]
```diff
@@ -148,6 +148,13 @@ class Table : public TableBase, public TableEntryEnumerable {
     /* Returns false if the key doesn't exists */
     virtual bool get(const std::string &key, std::vector<FieldValueTuple> &ovalues);
 
+    virtual bool getEntry(const std::string &key, const std::string &field,  std::string &value);
```
Comment (qiluo-msft) [inline common/table.h]: >getEntry [](start = 17, length = 8)

hget or getFieldValue
hset or setFiledValue

**Pull request merged**